From fd7fb616425fae4c0ea09f224d65e11fcb40be95 Mon Sep 17 00:00:00 2001
From: Erik Schumacher <erik.schumacher@iris-sensing.com>
Date: Thu, 13 Jun 2024 14:09:03 +0000
Subject: [PATCH] imx8-media-dev: Remove check and removal in probe

Due to different probing order this probe function will run at a
different time which leads to the wrong assumption that there are no
subdevices. By removing this section, the async notifier will be called
when the subdevices appear and the media linking will run.
---
 drivers/staging/media/imx/imx8-media-dev.c | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/drivers/staging/media/imx/imx8-media-dev.c b/drivers/staging/media/imx/imx8-media-dev.c
index 7c1ae75cb2f0..3a4a76e7a901 100644
--- a/drivers/staging/media/imx/imx8-media-dev.c
+++ b/drivers/staging/media/imx/imx8-media-dev.c
@@ -1107,21 +1107,6 @@ static int mxc_md_probe(struct platform_device *pdev)
 			dev_warn(&mxc_md->pdev->dev, "Sensor register failed\n");
 			goto clean_ents;
 		}
-
-		if (!mxc_md->link_status) {
-			if (mxc_md->valid_num_sensors > 0) {
-				ret = subdev_notifier_complete(&mxc_md->subdev_notifier);
-				if (ret < 0)
-					goto err_register_nf;
-
-				mxc_md_clean_unlink_channels(mxc_md);
-			} else {
-				/* no sensors connected */
-				mxc_md_unregister_all(mxc_md);
-				v4l2_async_nf_unregister(&mxc_md->subdev_notifier);
-				v4l2_async_nf_cleanup(&mxc_md->subdev_notifier);
-			}
-		}
 	}
 
 	return 0;
-- 
2.46.0

